pkg/workloads/cortex/tf_api/api.py

```diff
@@ -442,8 +442,8 @@ def download_dir_external(ctx, s3_path, local_path):
     util.mkdir_p(local_path)
     bucket_name, prefix = ctx.storage.deconstruct_s3_path(s3_path)
     storage_client = S3(bucket_name, client_config={})
-    objects = storage_client.list_objects(s3_path)
-    version = prefix.split("/")[-1]
+    objects = [obj[len(prefix) + 1 :] for obj in storage_client.search(prefix=prefix)]
```

Comment:
I think this trim will also not work correctly if the user includes the trailing "/". It might be safest to add the "/" to `prefix` (if it's not there already) before calling `search`, since then you'll also avoid the situation of matching prefixes which are not in the correct directory (e.g. if the user types in `1/2/3`, I think it would match a file with path `1/2/33/1` in addition to `1/2/3/1`.